scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2833,213 @@ cdef class Rotation:
+            raise ValueError("`weights` may not contain negative values")
+
+        # Check for an infinite weight, which indicates that the corresponding
+        # vector pair is the primary unmoving reference to which we align the
+        # other secondary vectors
+        weight_is_inf = np.isposinf(weights)
+        n_inf = np.sum(weight_is_inf)
+        if n_inf > 1:
+            raise ValueError("Only one infinite weight is allowed")
+
+        elif n_inf == 1:
+            if return_sensitivity:
+                raise ValueError("Cannot return sensitivity matrix with an "
+                                 "infinite weight or one vector pair")
+            a_pri = a[weight_is_inf]
+            b_pri = b[weight_is_inf]
+            a_pri /= _norm3(a_pri)
+            b_pri /= _norm3(b_pri)
+
+            # We first find the minimal angle rotation between the primary
+            # vectors.
+            cross = np.cross(b_pri, a_pri)
+            theta = np.arctan2(np.linalg.norm(cross), np.dot(a_pri[0], b_pri[0]))
+            if theta < 1e-3:  # small angle Taylor series approximation
+                theta2 = theta * theta
+                r = cross * (1 + theta2 / 6 + theta2 * theta2 * 7 / 360)
+            else:
+                r = cross * theta / np.sin(theta)
+            R_pri = Rotation.from_rotvec(r)
+
+            # Even with a single pair of vectors (~weight_is_inf is empty) the
+            # code below is correct and phi will be zero.
+            a_non_inf = a[~weight_is_inf]
+            b_non_inf = b[~weight_is_inf]
+            weights_non_inf = weights_adj[~weight_is_inf]
+
+            # We apply the first rotation to the b vectors to align the primary
+            # vectors, resulting in vectors c. The secondary vectors must now
+            # be rotated about that primary vector to best align c to a.
+            c_non_inf = R_pri.apply(b_non_inf)
+
+            # Calculate vector components for angle calculation
+            # Note that einsum('ij,ij->i', X, Y) is the row-wise dot product of
+            # X and Y.
+            S = np.einsum('ij,ij->i', np.cross(c_non_inf, a_non_inf), a_pri)
+            T = np.einsum('ij,ij->i', c_non_inf, a_non_inf)
+                - (np.einsum('ij,ij->i', c_non_inf, a_pri)
+                   * np.einsum('ij,ij->i', a_non_inf, a_pri))
+
+            # Calcuate the rotation angle about the primary axis
+            phi = np.arctan2(np.sum(weights_non_inf * S),
+                             np.sum(weights_non_inf * T))
+            R_sec = Rotation.from_rotvec(phi * a_pri)
+
+            # Compose these to get the optimal rotation
+            R_opt = R_sec * R_pri
+
+            # Calculated the root sum squared distance. The error will be 0 for
+            # the infinite weight vector, so we force that to avoid issues when
+            # multiplying by infinity
+            weights_inf_zero = weights.copy()
+            weights_inf_zero[weight_is_inf] = 0
```

Comment:
There is an issue here if the corresponding a and b vectors are not of equal length. In that instance the error will not be 0, and this will not line up with the baseline algorithm. I don't have a good idea of how to avoid this, without changing the behavior everywhere to calculate rssd from the normalized vectors.

Reply:
I agree with you. I think the correct view is not to consider the pair with the infinite weight as belonging to the cost function. It works as a constraint.

I'd say the only edge case is a single pair of vectors which can be viewed (sort of) either way. I think any solution will be fine:

1. Set `rssd` to zero and mention it in the docstring.
2. Compute it normally, but then forbid an infinite weight when only one pair of vectors is aligned (and perhaps adjust the code with artificial infinite weight).

Consider both options.

Reply:
I think setting `rssd` to 0 makes the most sense (and is the current behavior). For case 2, we still run into the same problem if the primary vectors are not of the same length, even if there are secondary vectors as well. For case 3, I think it would be rather confusing since I feel like the intuitive view is to see the weights as relative rather than absolute. So a weight of 1 would be in contrast to a weight of 0 and still be relatively infinite.

Added some explanation to the docstring that we force this component of rssd to be 0.

Reply:
That makes sense to me, and was not awkward to implement - only needed one if statement.